### PR TITLE
Part1

### PR DESCRIPTION
**Pull request by adeola003**
Kindly find below the detailed steps used in other to complete this project:
- I added @rushemaa as a collaborator to the repo;
- Tests have been installed;
- README has been updated;
- *jest* have has been installed;
- A file for tests has been created and tests for *add* and *remove* functions have been added;
- Due to tests not running, *babel & Jsdom* have been installed;
- Original functions have been refactored, and linters errors have been fixed after all tests passed.

### CODE REVIEW

File: src/functions.test.js [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+
+global.TextEncoder = require('util').TextEncoder;
+global.TextDecoder = require('util').TextDecoder;
+
+// eslint-disable-next-line
+const { JSDOM } = require('jsdom');
+
+const tasksArray = [];
+
+test('stringLength returns the correct length of a string', () => {
```
Comment (MahmoudBakr23) [inline src/functions.test.js]: - [ ] Please make sure to use the `describe` keyword in each of your tests to identify them. See the requirements [here](https://github.com/microverseinc/curriculum-javascript/blob/main/todo-list/tests_1.md#:~:text=Make%20sure%20you%20group%20your%20tests%20using%20the%20describe()%20method.).

File: src/functions.test.js [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+  const idToRemove = 0;
+  const expectedLength = 0;
+
+  // Act
+  removeTask(idToRemove);
+
+  // Assert
+  expect(tasksArray.length).toBe(expectedLength);
+  expect(tasksArray.find((task) => task.id === idToRemove)).toBeUndefined();
+});
```
Comment (MahmoudBakr23) [inline src/functions.test.js]: - [ ] Kindly write mock functions to test the DOM manipulation while adding and removing tasks and updating localstorage. See the requirements [here](https://github.com/microverseinc/curriculum-javascript/blob/main/todo-list/tests_1.md#:~:text=Mock%20a%20storage,in%20the%20DOM.).

**Review by MahmoudBakr23 — CHANGES_REQUESTED** [review]
Hello @adeola003 and @rushemaa 😃, 

Great work and effort you put into this project so far.

## Highlights 🚀 

There are some excellent points that you have made perfectly. Keep up! 💪
- Well done writing tests using `jest` for adding and removing tasks so far
- Descriptive PR and README file
- Correct Linters

However, there is always room for improvement.

> I'd like to remind you that your long hours of work are appreciated. However, the code review process is for improving your code, not criticizing it. 😄

 ## Required Changes ♻️
_Please check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better._

N/A

Cheers and Happy coding!👏👨‍💻💻

Feel free to mention @MahmoudBakr23 to leave any questions or comments in the PR thread if something is not 100% clear.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by Shisui6 — APPROVED** [review]
Hi @adeola003, 

# Approved 🪄
Your project is complete! There is nothing else to say other than... it's time to merge it 💎
Congratulations! 🎉 

## Highlights
- Tests are passing ✅

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you consider them as they can improve your code. Some of them were simply missed by the previous reviewer, and addressing them will really improve your application._

You can also consider the following:

[Optional] Please for future projects, kindly make sure your PR title is more descriptive

Cheers and Happy coding!🍻💻

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561), you have a limited number of reviews per project (check the exact number in your Dashboard). If you think the code review was unfair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**